src/AzureClient/Magic/ConnectMagic.cs

```diff
@@ -93,16 +103,41 @@ public ConnectMagic(IAzureClient azureClient)
         public override async Task<ExecutionResult> RunAsync(string input, IChannel channel)
         {
             var inputParameters = ParseInputParameters(input);
-
-            var storageAccountConnectionString = inputParameters.DecodeParameter<string>(ParameterNameStorageAccountConnectionString);
-            if (string.IsNullOrEmpty(storageAccountConnectionString))
+            if (!inputParameters.Any())
             {
                 return await AzureClient.GetConnectionStatusAsync(channel);
             }
 
-            var subscriptionId = inputParameters.DecodeParameter<string>(ParameterNameSubscriptionId);
-            var resourceGroupName = inputParameters.DecodeParameter<string>(ParameterNameResourceGroupName);
-            var workspaceName = inputParameters.DecodeParameter<string>(ParameterNameWorkspaceName);
+            var resourceId = inputParameters.DecodeParameter<string>(ParameterNameResourceId, defaultValue: string.Empty);
+            var subscriptionId = string.Empty;
+            var resourceGroupName = string.Empty;
+            var workspaceName = string.Empty;
+
+            // A valid resource ID looks like:
+            // /subscriptions/f846b2bd-d0e2-4a1d-8141-4c6944a9d387/resourceGroups/RESOURCE_GROUP_NAME/providers/Microsoft.Quantum/Workspaces/WORKSPACE_NAME
+            var resourceIdRegex = new Regex(
+                @"^\/subscriptions\/([a-zA-Z0-9\-]*)\/resourceGroups\/([^\s\/]*)\/providers\/Microsoft\.Quantum\/Workspaces\/([^\s\/]*)$");
+            if (!string.IsNullOrEmpty(resourceId) && resourceIdRegex.IsMatch(resourceId))
+            {
+                // match.Groups will be a GroupCollection containing four Group objects:
+                // -> match.Groups[0]: The full resource ID for the Azure Quantum workspace
+                // -> match.Groups[1]: The Azure subscription ID
+                // -> match.Groups[2]: The Azure resource group name
+                // -> match.Groups[3]: The Azure Quantum workspace name
+                var match = resourceIdRegex.Match(resourceId);
+                subscriptionId = match.Groups[1].Value;
+                resourceGroupName = match.Groups[2].Value;
+                workspaceName = match.Groups[3].Value;
+            }
+            else
+            {
+                // look for each of the parameters individually
+                subscriptionId = inputParameters.DecodeParameter<string>(ParameterNameSubscriptionId, defaultValue: string.Empty);
+                resourceGroupName = inputParameters.DecodeParameter<string>(ParameterNameResourceGroupName, defaultValue: string.Empty);
+                workspaceName = inputParameters.DecodeParameter<string>(ParameterNameWorkspaceName, defaultValue: string.Empty);
+            }
+
+            var storageAccountConnectionString = inputParameters.DecodeParameter<string>(ParameterNameStorageAccountConnectionString, defaultValue: string.Empty);
```

Comment:
So the resource ID doesn't include the storage string, but if there is no linked storage, it can be specified like `%azure.connect resourceId=... storage=...`? It looks like that particular combination isn't included in the documentation or tests, should it be added?

Reply:
Yes, good point. From what I understand, the common case should have linked storage, but it would still be good to document the case where it doesn't, and even more important to test it.